app/src/main/java/com/x8bit/bitwarden/data/vault/datasource/sdk/VaultSdkSourceImpl.kt

```diff
@@ -432,6 +449,119 @@ class VaultSdkSourceImpl(
+                                options: CheckUserOptions,
+                                newCredential: Fido2CredentialNewView,
+                            ): CipherViewWrapper {
+                                checkUserAndPickCredentialForCreation(options, newCredential)
+                                return CipherViewWrapper(selectedCipherView)
+                            }
+
+                            override suspend fun isVerificationEnabled() = isVerificationSupported
+
+                            override suspend fun pickCredentialForAuthentication(
+                                availableCredentials: List<CipherView>,
+                            ): CipherViewWrapper {
+                                // This delegate should never be invoked during credential
+                                // registration. If it is we throw an exception to the SDK so that
+                                // a spec compliant error can be returned from `register`.
+                                throw UnsupportedOperationException(
+                                    "Pick Credential not supported during passkey registration.",
+                                )
+                            }
+                        },
+                        credentialStore = object : Fido2CredentialStore {
+                            override suspend fun findCredentials(
+                                ids: List<ByteArray>?,
+                                ripId: String,
+                            ): List<CipherView> =
+                                when (val result = findCredentials(ids.orEmpty(), ripId)) {
+                                    is FindFido2CredentialsResult.Error -> {
+                                        // This exception is caught and handled by the SDK, which
+                                        // uses it to produce a FIDO 2 spec compliant attestation
+                                        // error.
+                                        throw CreateCredentialUnknownException()
+                                    }
+
+                                    is FindFido2CredentialsResult.Success -> {
+                                        result.cipherViews
+                                    }
+                                }
+
+                            override suspend fun saveCredential(cred: Cipher) {
+                                if (saveCredential(cred) is SaveCredentialResult.Error) {
+                                    // This exception is caught and handled by the SDK, which uses
+                                    // it to produce a FIDO 2 spec compliant attestation error.
+                                    throw CreateCredentialUnknownException()
+                                }
+                            }
+
+                            override suspend fun allCredentials(): List<CipherView> = cipherViews
+                        },
```

Comment:
🤔 I think `Fido2UserInterface` and `Fido2CredentialStore` should be extracted to specific objects so they can be reused on the different Fido2 flows and we can save the checks on whether the delegate should/shouldn't be called on each flow like `"Pick Credential not supported during passkey registration"`. I think we should just let the SDK handle the appropriate flow.